accord-core/src/main/java/accord/primitives/Routables.java

```diff
@@ -48,21 +48,21 @@ enum Slice
 
     boolean isEmpty();
     boolean intersects(AbstractRanges<?> ranges);
-    boolean intersects(AbstractKeys<?, ?> keys);
-    default boolean intersects(Routables<?, ?> routables)
+    boolean intersects(AbstractKeys<?> keys);
+    default boolean intersects(Routables<?> routables)
     {
         switch (routables.domain())
         {
             default: throw new AssertionError();
-            case Key: return intersects((AbstractKeys<?, ?>) routables);
+            case Key: return intersects((AbstractKeys<?>) routables);
             case Range: return intersects((AbstractRanges<?>) routables);
         }
     }
     boolean contains(RoutableKey key);
-    boolean containsAll(Routables<?, ?> keysOrRanges);
+    boolean containsAll(Routables<?> keysOrRanges);
 
-    U slice(Ranges ranges);
-    Routables<K, ?> slice(Ranges ranges, Slice slice);
+    Routables slice(Ranges ranges);
```

Comment:
```suggestion
    Routables<?> slice(Ranges ranges);
```

the `<?>` is to avoid a compiler warning... all subtypes override and define a more specific type

Reply:
:+1: I think this is due to a bug in IntelliJ's refactoring of generic types.